src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs

```diff
@@ -88,7 +88,8 @@ internal static string GetDosDeviceForNetworkPath(char deviceName)
                         }
                         else
                         {
-                            Diagnostics.Assert(false, "Really it is a dead code since GetDosDevice() is called only if PSDrive.DriveType == DriveType.Network");
+                            // TODO: This actually hit in a Parallels VM, so it's not quite dead yet!
+                            // Diagnostics.Assert(false, "Really it is a dead code since GetDosDevice() is called only if PSDrive.DriveType == DriveType.Network");
```

Comment:
This is a fallback from another p/invoke GetUNCForNetworkDrive - it is main code path. This says that something can work incorrectly on Parallels VM.
So I suggest investigate the issue in separate PR.